src/redocly/index.js

```diff
@@ -0,0 +1,161 @@
+
+import query from './query';
+
+export default class RedoclyClient {
+  constructor() {
+    const credentialsPath = resolve(homedir(), '.redocly.token.json');
+    let credentials = {};
+    if (existsSync(credentialsPath)) {
+      credentials = JSON.parse(readFileSync(credentialsPath, 'utf-8'));
+    }
+    const authDetails = RedoclyClient.authorize(credentials.token);
+    if (authDetails) {
+      this.accessToken = credentials.token;
+    }
+  }
+
+  getAuthorizationHeader() {
+    return this.accessToken;
+  }
+
+  login(accessToken) {
+    const credentialsPath = resolve(homedir(), '.redocly.token.json');
+    const authDetails = RedoclyClient.authorize(accessToken);
+    if (authDetails) {
+      this.accessToken = accessToken;
+      const credentials = {
+        token: accessToken,
+      };
+      writeFileSync(credentialsPath, JSON.stringify(credentials));
+      process.stdout.write('Authorization confirmed.\n');
+    } else {
+      process.stdout.write('Authorization failed. Please check if you entered a valid token.\n');
+    }
+  }
+
+  logout() {
+    const credentialsPath = resolve(homedir(), '.redocly.token.json');
+    if (existsSync(credentialsPath)) {
+      unlinkSync(credentialsPath);
+    }
+  }
+
+  query(queryString, parameters = {}, headers = {}) {
+    return query(queryString, parameters,
+      {
+        Authorization: this.accessToken,
+        ...headers,
+      });
+  }
+
+  static authorize(accessToken, verbose = false) {
+    try {
+      const result = query(`
+      {
+        definitions {
+          id
+        }
+      }
+      `,
+      {},
+      {
+        Authorization: accessToken,
+      });
+      return result;
+    } catch (e) {
+      if (verbose) process.stderr.write(e);
+      return null;
+    }
+  }
+
+  isLoggedIn() {
+    try {
+      this.query(`
+        {
+          viewer {
+            id
+            email
+          }
+        }
+      `);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  listDefinitions() {
+    const result = this.query(`
+    {
+      definitions{
+        id
+        definitionVersions{
+          id
+          name
+          sourceType
+        }
+      }
+    }
+    `,
+    {});
+    return result.definitions;
+  }
+
+  updateDependencies(dependencies, authorizationToken) {
+    const r = this.query(`
+    mutation UpdateBranchDependencies ($dependencies: [String!]!, $definitionId: Int!, $versionId: Int!, $branchId: Int!) {
+      updateBranchDependencies(definitionId:$definitionId, versionId:$versionId, branchId:$branchId, dependencies:$dependencies){
+        branchName
+      }
+    }
+    `,
+    {
+      dependencies: dependencies || [],
+      definitionId: parseInt(process.env.DEFINITION, 10),
+      versionId: parseInt(process.env.VERSION, 10),
+      branchId: parseInt(process.env.BRANCH, 10),
+    }, {
+      Authorization: authorizationToken,
+    });
+    return r;
+  }
+
+  processRegistryDependency(link, ctx) {
```

Comment:
I would rename this method. The name implies that link is registry dependency but we check this in this method. And we do not process.

I would rename this method to `isRegistryURL` and push to context not in this method. This class should better not know about ctx. 

Also, make this method `static`